### PR TITLE
16 Search Option

### PR DESCRIPTION
**Pull request by Arieldelossantos**
# **What is new?**

### Search option.

 Search by name through developers list.

## *Visual Aids*

|Android|iOS|
|--|--|
|searchDemo|Demo_Search|

### CODE REVIEW

File: DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+        void AddSearchToToolbar(string pageTitle)
+        {
+            if (GetToolbar() is Toolbar toolBar
+                && toolBar.Menu?.FindItem(Resource.Id.ActionSearch)?.ActionView?.JavaCast<SearchView>().GetType() != typeof(SearchView))
+            {
+                toolBar.Title = pageTitle;
+                toolBar.InflateMenu(Resource.Menu.MainMenu);
+
+                if (toolBar.Menu?.FindItem(Resource.Id.ActionSearch)?.ActionView?.JavaCast<SearchView>() is SearchView searchView)
```
Comment (RaulMonteroC) [inline DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs]: Add this condition as a variable and reference it on the If statement to make the intent clearer. 

Something like:

``
var searchView = toolBar.Menu?.FindItem(Resource.Id.ActionSearch)?.ActionView?.JavaCast<SearchView>() as SearchView
 if (searchView != null)
``

File: DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    searchView.ImeOptions = (int)ImeAction.Search;
+                    searchView.InputType = (int)InputTypes.TextVariationFilter;
+                    searchView.SetIconifiedByDefault(false);
+
+                    searchView.SetBackgroundResource(Resource.Drawable.round_searchview);
+
+                    var _closeBtn =  searchView.FindViewById<Android.Widget.ImageView>(Resource.Id.search_close_btn);
+                    var _searchBtn = searchView.FindViewById<Android.Widget.ImageView>(Resource.Id.search_mag_icon);
+                    _closeBtn.SetColorFilter(Color.Black.ToAndroid());
+                    _searchBtn.SetColorFilter(Color.Black.ToAndroid());
```
Comment (RaulMonteroC) [inline DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs]: Extract this to a private method called **ConfigureSearchView** or something along those lines

File: DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                }
+            }
+        }
+
+        void HandleQueryTextChange(object sender, SearchView.QueryTextChangeEventArgs e)
```
Comment (RaulMonteroC) [inline DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs]: Add **explicit** private access modifier for this method and any other that doesn't have it.

File: DominicanWhoCodes/Modules/Developers/ViewModels/DevelopersViewModel.cs [hunk 4]
```diff
@@ -25,11 +25,15 @@ public class DevelopersViewModel : BaseViewModel
...
+        private IEnumerable<Developer> _developersContainer;
+
+        public ObservableCollection<Developer> Developers { get; } = new ObservableCollection<Developer>();
 
         [Reactive]
         public Developer SelectedDeveloper { get; set; }
 
-        public ObservableCollection<Developer> Developers { get; } = new ObservableCollection<Developer>();
+        [Reactive]
+        public string DeveloperName { get; set; }
```
Comment (RaulMonteroC) [inline DominicanWhoCodes/Modules/Developers/ViewModels/DevelopersViewModel.cs]: This name is misleading, change it to SearchText or something similar.

**Review by RaulMonteroC — CHANGES_REQUESTED** [review]

File: DominicanWhoCodes/Modules/Developers/ViewModels/DevelopersViewModel.cs (hunk 4 above)
Comment (Arieldelossantos) [reply DominicanWhoCodes/Modules/Developers/ViewModels/DevelopersViewModel.cs]: done

File: DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs (hunk 3 above)
Comment (Arieldelossantos) [reply DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs]: done

File: DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs (hunk 2 above)
Comment (Arieldelossantos) [reply DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs]: done

File: DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs (hunk 1 above)
Comment (Arieldelossantos) [reply DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs]: the way it has been done, is how compiler suggest :/

Comment (RaulMonteroC) [reply DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs]: I know, but it doesn't matter.

We should strive for simplicity and readability, always.

Comment (Arieldelossantos) [reply DominicanWhoCodes.Android/Renderers/SearchPageRenderer.cs]: done

**Review by RaulMonteroC — APPROVED** [review]

**Pull request merged**